core/src/main/scala/kafka/server/ControllerApis.scala

```diff
@@ -74,7 +75,15 @@ class ControllerApis(val requestChannel: RequestChannel,
 
   val authHelper = new AuthHelper(authorizer)
   val requestHelper = new RequestHandlerHelper(requestChannel, quotas, time)
-  private val aclApis = new AclApis(authHelper, authorizer, requestHelper, "controller", config)
+  private[server] val aclApis = new AclApis(authHelper, authorizer, requestHelper, "controller", config)
+  private val _isClosed = new AtomicBoolean(false)
+
+  def isClosed: Boolean = _isClosed.get()
+
+  def close(): Unit = if (_isClosed.compareAndSet(false, true)) {
```

Comment:
Why do we need the compare and set here and the other class?

Reply:
make sure it gets closed only once. I'm fine to remove it if it is overkill :)